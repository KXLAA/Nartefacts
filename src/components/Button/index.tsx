import * as S from './styles'
import { capitalize } from 'utils/utils'

export type ButtonProps = {
  text?: string
  buttonColor?: string
  textColor?: string
}

export const Button: React.FC<ButtonProps> = ({
  buttonColor,
  text,
  textColor,
}) => {
  return (
    <S.Button buttonColor={buttonColor} textColor={textColor} role="button">
      {capitalize(text!) || 'Button'}
    </S.Button>
  )
}
