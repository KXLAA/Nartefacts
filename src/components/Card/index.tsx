import Image from 'next/image'

import * as S from '@/components/Card/styles'

export type CardProps = {
  image?: string | null
  colors?: string[] | null
  likeCount?: number | null
  textColor?: string | null
  albumArt?: string | null
  spotify?: string | null
  apple?: string | null
  type?: string | null
  description?: string | null
  title?: string | null
  artist?: {
    name?: string | null
    photoUrl?: string | null
    biography?: string | null
  } | null
}

export const Card: React.FC<CardProps> = ({ albumArt }) => {
  return (
    <S.Card>
      <S.Gradient />
      <S.ImageWrapper>
        <Image
          src={albumArt!}
          height={385}
          width={385}
          alt={'album art'}
          layout="responsive"
        />
      </S.ImageWrapper>
    </S.Card>
  )
}
