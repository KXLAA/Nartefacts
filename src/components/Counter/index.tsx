import * as S from './styles'

export type CounterProps = {
  count?: number | undefined | null
}

export const Counter: React.FC<CounterProps> = ({ count }) => {
  return (
    <S.Wrapper>
      <S.Counter>
        <p>{`${count} Gradients Generated`} </p>
      </S.Counter>
    </S.Wrapper>
  )
}
