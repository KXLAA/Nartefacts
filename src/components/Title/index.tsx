import * as S from '@/components/Title/styles'

export type TitleProps = {
  text?: string
}

export const Title: React.FC<TitleProps> = ({ text }) => {
  return (
    <S.Wrapper>
      <p>{text || 'description'}</p>
    </S.Wrapper>
  )
}
