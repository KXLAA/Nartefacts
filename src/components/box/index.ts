export { Box } from './Box'
export type { BoxProps } from './box.types'
