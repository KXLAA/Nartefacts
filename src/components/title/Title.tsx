import { StyledWrapper } from './styles'
import { TitleProps } from './types'

export const Title: React.FC<TitleProps> = ({ text }) => {
  return (
    <StyledWrapper>
      <p>{text || 'description'}</p>
    </StyledWrapper>
  )
}
