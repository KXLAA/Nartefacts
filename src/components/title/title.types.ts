export type TitleProps = {
  text?: string
  children?: React.ReactNode
}
