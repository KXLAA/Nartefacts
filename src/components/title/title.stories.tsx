import { Meta, Story } from '@storybook/react'

import { Title } from '@/components/title'

export default {
  component: Title,
  title: 'Title',
} as Meta

export const Basic: Story = () => <Title />
