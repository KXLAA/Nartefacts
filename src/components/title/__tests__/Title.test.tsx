import { render } from '@testing-library/react'

import { Title } from '@/components/title'

describe('<Title />', () => {
  it('should render the with default values', () => {
    const { getByText } = render(<Title />)
    const description = getByText(/description/)
    expect(description).toBeInTheDocument()
  })

  it('should render <Title /> correctly', () => {
    const { container } = render(<Title />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the with correctly with given props', () => {
    const { getByText } = render(<Title text="property" />)
    const description = getByText(/property/)
    expect(description).toBeInTheDocument()
  })
})
