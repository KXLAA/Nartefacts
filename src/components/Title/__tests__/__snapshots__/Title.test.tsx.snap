// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> should render the entire title correctly 1`] = `
.c0 {
  width: 100%;
  background-color: #202020;
  padding: 32px;
  border: 1px solid #5A5A5A;
  border-radius: 8px;
  text-align: center;
  font-weight: bold;
}

<div
  class="c0"
>
  <p>
    description
  </p>
</div>
`;
